### PR TITLE
Introduce HTML guide

### PR DESCRIPTION
**Pull request by dgalarza**

### CODE REVIEW

**Review by tute — APPROVED** [review]

**Review by tysongach — APPROVED** [review]

File: html/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# HTML
+
+- Prefer input `type=text` paired with the [`inputmode`] attribute, to `type=number`.
```
Comment (calebhearth) [inline html/README.md:3]: Why not both?

Comment (tute) [reply html/README.md:3]: See here https://buoy-software.slack.com/archives/C04GXBNK9BJ/p1699375450341469?thread_ts=1699375336.950699&cid=C04GXBNK9BJ

Comment (calebhearth) [reply html/README.md:3]: Tried out both of these and they appear to give the same result, but `[type=number]` is still more semantic:
```html
  <label for="text-numeric">type=text + inputmode=numeric</label>
  <input type="text" inputmode="numeric" id="text-numeric" />
  <label for="numeric-number">type=number + inputmode=numeric</label>
  <input type="number" inputmode="numeric" id="numeric-number" />
```
https://github.com/BuoySoftware/guides/assets/711717/023fbffd-e7a7-4c66-8c4b-79ccc6c92a1e

Comment (tute) [reply html/README.md:3]: I seem to remember some browsers incremented decimal points, but that may be in the past now (capybara does, I remember having trouble while testing).

Tyson quotes accessibility issues and unwanted inc/decrement arrows.

**Review by charlesgwilson — APPROVED** [review]

File: html/README.md (hunk 1 above)
Comment (tysongach) [reply html/README.md:3]: Right - `type=number` renders those undesired increment/decrement arrows, particularly on desktop browsers…

Screenshot 2023-11-09 at 12 58 44

Generally speaking, those steppers are not useful to users and can often time clutter up the UI/get in the way.

As for semantics, `text` inputs are quite generic (I read it as any type of text, numbers included), so I think they still work in lots of places. In my opinion, the input’s label and any hints (if necessary) matter more and should be highly descriptive as to what type of value we want users to input. 

Screen readers will typically announce `type=number` inputs as ‘stepper’ or ‘increment’ inputs, potentially adding confusion.

This [great article gives a nice checklist for when a number input should probably be used](https://adamsilver.io/blog/form-design-when-to-use-the-number-input/):

> - incrementing and decrementing makes sense
> - the number doesn’t have a leading zero
> - the value doesn’t contain letters, slashes, minus signs, and decimal points.
> - the number isn’t very large

Perhaps we expand our guide addition here to include these details?

Comment (dgalarza) [reply html/README.md:3]: Thanks @tysongach! I've updated to include a link to "Form design: when to use the number input"

**Pull request merged**